server/odc-service/src/main/java/com/oceanbase/odc/service/structurecompare/StructureComparisonTask.java

```diff
@@ -0,0 +1,160 @@
+import com.oceanbase.odc.service.objectstorage.model.ObjectMetadata;
+import com.oceanbase.odc.service.structurecompare.model.DBObjectComparisonResult;
+import com.oceanbase.odc.service.structurecompare.model.DBStructureComparisonConfig;
+import com.oceanbase.tools.dbbrowser.model.DBObjectType;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author jingtian
+ * @date 2024/1/18
+ * @since ODC_release_4.2.4
+ */
+@Slf4j
+public class StructureComparisonTask implements Callable<DBStructureComparisonTaskResult> {
+    private DBStructureComparisonConfig srcConfig;
+    private DBStructureComparisonConfig tgtConfig;
+    private Long taskId;
+    private User user;
+    private StructureComparisonRepository structureComparisonRepository;
+    private StructureComparisonTaskRepository structureComparisonTaskRepository;
+    private ObjectStorageFacade objectStorageFacade;
+    private DefaultDBStructureComparator comparator = new DefaultDBStructureComparator();
+    @Getter
+    private volatile DBStructureComparisonTaskResult taskResult =
+            new DBStructureComparisonTaskResult(TaskStatus.PREPARING);
+    private static final String STRUCTURE_COMPARISON_RESULT_FILE_NAME = "structure-comparison-result.sql";
+    private final String STRUCTURE_COMPARISON_BUCKET_NAME = "structure-comparison";
+
+    public StructureComparisonTask(@NonNull DBStructureComparisonConfig srcConfig,
+            @NonNull DBStructureComparisonConfig tgtConfig,
+            @NonNull Long taskId,
+            @NonNull Long structureComparisonTaskId,
+            @NonNull User user,
+            @NonNull StructureComparisonTaskRepository structureComparisonTaskRepository,
+            @NonNull StructureComparisonRepository structureComparisonRepository,
+            @NonNull ObjectStorageFacade objectStorageFacade) {
+        this.srcConfig = srcConfig;
+        this.tgtConfig = tgtConfig;
+        initTaskResult(structureComparisonTaskId);
+        this.taskId = taskId;
+        this.user = user;
+        this.structureComparisonRepository = structureComparisonRepository;
+        this.structureComparisonTaskRepository = structureComparisonTaskRepository;
+        this.objectStorageFacade = objectStorageFacade;
+    }
+
+    private void initTaskResult(Long structureComparisonTaskId) {
+        this.taskResult.setTaskId(structureComparisonTaskId);
+        if (!srcConfig.getBlackListMap().isEmpty()) {
+            this.taskResult.setComparingList(srcConfig.getBlackListMap().get(DBObjectType.TABLE).stream()
+                    .map(tableName -> new Comparing(DBObjectType.TABLE, tableName))
+                    .collect(Collectors.toList()));
+        }
+    }
+
+    @Override
+    public DBStructureComparisonTaskResult call() throws Exception {
+        SecurityContextUtils.setCurrentUser(user);
+        StructureComparisonTraceContextHolder.trace(user.getId(), taskId);
```

Comment:
Why not use TraceContextHolder

Reply:
because structural comparison tasks need to configure a separate directory for storing logs, link to `String TASK_WORK_SPACE = "structureComparisonWorkSpace";`. I see other tasks have similar configurations such as `PartitionPlanTaskTraceContextHolder`, `ShadowTaskTraceContextHolder`